### PR TITLE
IIA-43-Log-out+Login Screen constraints

### PR DESCRIPTION
**Pull request by colac-carla-30126**

### CODE REVIEW

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 1]
```diff
@@ -38,5 +44,43 @@ extension ProfileInteractor: ProfileInteractorInput {
...
+
+        resultDeleteSession(sessionEntity: sessionEntity) {
+            deleteSession in
+            guard let deleteSession = deleteSession,
+                  deleteSession.success ?? false else { return }
+        }
+    }
+    
+    func resultDeleteSession(sessionEntity: String, completion: @escaping (DeleteSessionResult?) -> Void) {
+        let url = URL(string: "\(TheMovieDBApiConstants.baseUrl)/authentication/session?api_key\(TheMovieDBApiConstants.apiKey)")!
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:65]: use guard let and please avoid force unwrapping

File: EnMovie/Modules/Profile/ProfileBuilder.swift [hunk 2]
```diff
@@ -11,15 +11,30 @@ import UIKit
 
 protocol ProfileDependency: Dependency {}
 
+final class ProfilComponent: Component<ProfileDependency> {
+    
+}
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileBuilder.swift:16]: you can keep it on one line

File: EnMovie/Modules/Profile/ProfilePresenter.swift [hunk 3]
```diff
@@ -20,9 +21,24 @@ final class ProfilePresenter {
     }
 }
 
+extension ProfilePresenter: ProfileInput {
+    func logout() {
+        interactor.logout()
+    }
+    
+
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfilePresenter.swift:29]: remove unnecessary new lines

File: EnMovie/Modules/Profile/ProfileViewController.swift [hunk 4]
```diff
@@ -23,6 +23,7 @@ protocol ProfileViewOutput {
     func viewIsReady()
     func didSelectCell(at row: Int)
     func cell(for row: Int) -> Settings?
+    func logout()
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileViewController.swift:27]: new line

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 5]
```diff
@@ -38,5 +44,43 @@ extension ProfileInteractor: ProfileInteractorInput {
...
+            completion(nil)
+            return
+        }
+        request.httpMethod = "DELETE"
+        request.httpBody = encoded
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+
+        let task = URLSession.shared.dataTask(with: request, completionHandler: { data, response, error in
+            do {
+                let result = try JSONDecoder().decode(DeleteSessionResult.self, from: data!)
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:77]: avoid force unwrapping

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 6]
```diff
@@ -38,5 +44,43 @@ extension ProfileInteractor: ProfileInteractorInput {
         return settingsModal
     }
     
+    func logout() {
+
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:48]: remove new line

File: EnMovie/Modules/Profile/ProfileBuilder.swift [hunk 7]
```diff
@@ -11,15 +11,30 @@ import UIKit
...
     func build() -> UIViewController
 }
 
+public protocol ProfileInput {
+    func logout()
+}
+
 final class ProfileBuilder: Builder<ProfileDependency>, ProfileBuildable {
     func build() -> UIViewController {
+        let window = UIWindow()
```
Comment (iOSinternshipen) [inline EnMovie/Modules/Profile/ProfileBuilder.swift:31]: tbh it's not really a good approach to create a new window, maybe you can use the one from UIApplication

https://stackoverflow.com/questions/3782066/how-to-get-my-uiwindow-using-uiapplication

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 8]
```diff
@@ -38,5 +44,44 @@ extension ProfileInteractor: ProfileInteractorInput {
         return settingsModal
     }
     
+    func logout() {
+        
+        UserDefaults.standard.removeObject(forKey: "sessionTokenKey")
+        UserDefaults.standard.removeObject(forKey: "token")
+        UserDefaults.standard.removeObject(forKey: "account_id")
+        self.output?.logoutSuccessfull()
```
Comment (iOSinternshipen) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:52]: self not needed

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 9]
```diff
@@ -38,5 +44,44 @@ extension ProfileInteractor: ProfileInteractorInput {
...
+        
+        guard let sessionEntity = UserDefaults.standard.string(forKey: "session_id") else { return }
+        UserDefaults.standard.removeObject(forKey: "session_id")
+
+        resultDeleteSession(sessionEntity: sessionEntity) {
+            deleteSession in
```
Comment (iOSinternshipen) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:58]: same row with result pleas etc respect the guidelines

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 10]
```diff
@@ -38,5 +44,44 @@ extension ProfileInteractor: ProfileInteractorInput {
...
+            guard let deleteSession = deleteSession,
+                  deleteSession.success ?? false else { return }
```
Comment (iOSinternshipen) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:60]: do you need guard? I mean it seems that you don't show anything after or before

File: EnMovie/Modules/Profile/ProfilePresenter.swift [hunk 11]
```diff
@@ -20,9 +21,24 @@ final class ProfilePresenter {
     }
 }
 
+extension ProfilePresenter: ProfileInput {
+    func logout() {
+        interactor.logout()
+    }
+
```
Comment (iOSinternshipen) [inline EnMovie/Modules/Profile/ProfilePresenter.swift:28]: lines not needed

File: EnMovie/Modules/Profile/ProfileInteractor.swift [hunk 12]
```diff
@@ -38,5 +44,43 @@ extension ProfileInteractor: ProfileInteractorInput {
...
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+
+        let task = URLSession.shared.dataTask(with: request, completionHandler: { data, response, error in
+            do {
+                let result = try JSONDecoder().decode(DeleteSessionResult.self, from: data!)
+                completion(result)
+            } catch {
+                print("JSON decoder error: \(error)")
+            }
+            print("Error: \(String(describing: error))")
```
Comment (evakiraly) [inline EnMovie/Modules/Profile/ProfileInteractor.swift:82]: maybe it would be nicer if you don't use print for error handling

**Review by iOSinternshipen — APPROVED** [review]
I hope it works :))

**Review by evakiraly — APPROVED** [review]

**Pull request merged**